### PR TITLE
feat: Print folder sizes with cypress cache list command.

### LINKED ISSUE #6404: Print folder sizes with "cypress cache list" command
To improve "cypress cache list" in v4, which right now just prints list of versions

```
$ npx cypress cache list
3.0.1, 3.0.2, 3.0.3, 3.1.0, 3.1.1, 3.1.2, 3.1.3, 3.1.4, 3.1.5, 3.2.0, 3.3.0, 3.3.1, 3.3.2, 3.4.0, 3.4.1, 3.5.0, 3.6.0, 3.6.1, 3.7.0, 3.8.0, 3.8.1, 3.8.2, 3.8.3, 3.8.4, 4.0.0, 4.0.1
```

It would be nice to print a table with version number, folder size, maybe last access time (today, 2 days ago, a month ago, etc)

### PR DESCRIPTION
**Pull request by sainthkh**
* Closes #6404

### User facing changelog
You can list the cypress cache folder sizes with `cypress cache list --size` command. 

I added `--size` as an option here because it makes command too slow. -> More about the implementation details below. 

#### Why was this change necessary?
Some users might want to check the size of cypress cache folders. 

#### What is affected by this change?
N/A

#### Any implementation details to explain?
1. At first, I used [`get-folder-size`](https://www.npmjs.com/package/get-folder-size) library. But it is dependent on `tiny-each-async` and `gar`. Maybe because of them, it slows down when I used over 10 cypress caches. And it didn't use the promise. So, I decided to create our own get-folder-size.
1. And this `getFolderSize` function recursively loops the folder and add the sizes of the files. Because of that, it's really slow. That's why I added it as `--size` option.
1. Finally, `getFolderSize` returns actual folder size, not the allocated disk space.

### How has the user experience changed?
Screenshot from 2020-09-16 14-42-33

### PR Tasks
* [x] Have tests been added/updated?
* [x] Has the original issue been tagged with a release in ZenHub? 
* [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? => https://github.com/cypress-io/cypress-documentation/pull/3217

### CODE REVIEW

**Comment by sainthkh** [comment]
Flaky failure.

File: cli/lib/tasks/cache.js [hunk 1]
```diff
@@ -29,25 +31,38 @@ const clear = () => {
...
+      head,
     })
 
     binaries.forEach((binary) => {
       const versionString = colors.values(binary.version)
       const lastUsed = binary.accessed ? colors.dates(binary.accessed) : 'unknown'
+      const row = [versionString, lastUsed]
+
+      if (showSize) {
+        const size = colors.size(`${(binary.size / 1024 / 1024).toFixed(1)}MB`)
```
Comment (bahmutov) [inline cli/lib/tasks/cache.js:53]: move this into a utility function

File: cli/lib/tasks/get-folder-size.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const fs = require('../fs')
+const { join } = require('path')
+const Bluebird = require('bluebird')
+
+async function getSize (path) {
```
Comment (bahmutov) [inline cli/lib/tasks/get-folder-size.js:5]: Could you add JSDoc comment to this function?

File: cli/test/lib/tasks/cache_spec.js [hunk 3]
```diff
@@ -204,5 +204,21 @@ describe('lib/tasks/cache', () => {
...
       await snapshotWithHtml('second-binary-never-used.html')
     })
+
+    it('shows sizes', async function () {
+      sinon.stub(state, 'getPathToExecutable').returns('/.cache/Cypress/1.2.3/app/cypress')
+
+      const statAsync = sinon.stub(fs, 'statAsync')
+
+      statAsync.onFirstCall().resolves({
+        atime: moment().subtract(3, 'month').valueOf(),
```
Comment (bahmutov) [inline cli/test/lib/tasks/cache_spec.js:214]: could you add an actual stats object to these mock folders, maybe even a recursive file to actually get folder size and not just 0.0?

**Review by bahmutov — CHANGES_REQUESTED** [review]
a few tests are needed, also see inline comments

File: cli/lib/tasks/cache.js (hunk 1 above)
Comment (sainthkh) [reply cli/lib/tasks/cache.js]: Done.

File: cli/lib/tasks/get-folder-size.js (hunk 2 above)
Comment (sainthkh) [reply cli/lib/tasks/get-folder-size.js:14]: Done.

File: cli/test/lib/tasks/cache_spec.js (hunk 3 above)
Comment (sainthkh) [reply cli/test/lib/tasks/cache_spec.js:219]: Done.

**Comment by sainthkh** [comment]
Flaky failures.

**Review by bahmutov — APPROVED** [review]
Super, verified locally that it works with `node ./bin/cypress cache list --size` and will try rerunning the pipeline until it is green before merging. I will open documentation PR

**Comment by Saibamen** [comment]
Will be option to delete cache for previous Cypress versions?

**Comment by bahmutov** [comment]
@Saibamen no, that's a separate feature from https://github.com/cypress-io/cypress/issues/5972

**Comment by jennifer-shehane** [comment]
@sainthkh There's an issue from this PR: https://github.com/cypress-io/cypress/issues/8856

**Pull request merged**